src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefLocalTests.cs

```diff
@@ -1669,6 +1669,366 @@ .maxstack  2
   IL_000d:  stloc.0
   IL_000e:  ret
 }");
+
+            CompileAndVerify(text, options: TestOptions.ReleaseDll).VerifyIL("Program.M()", @"
+{
+  // Code size       14 (0xe)
+  .maxstack  2
+  IL_0000:  ldc.i4.1
+  IL_0001:  newarr     ""int""
+  IL_0006:  ldc.i4.0
+  IL_0007:  ldelema    ""int""
```

Comment:
>IL_0007:  ldelema    ""int"" [](http://example.com/codeflow?start=2&length=28)

Is there a change in behavior for this scenario? I think it is not necessary because when array element is a value type it is not subject for variance and type checks are not necessary.